velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -1697,4 +1697,129 @@ std::shared_ptr<DateTimeFormatter> buildJodaDateTimeFormatter(
+        cur += 2;
+      } else {
+        // Case 2: find closing single quote
+        int64_t count = numLiteralChars(startTokenPtr + 1, end);
+        if (count == -1) {
+          VELOX_USER_FAIL("No closing single quote for literal");
+        } else {
+          for (int64_t i = 1; i <= count; i++) {
+            builder.appendLiteral(startTokenPtr + i, 1);
+            if (*(startTokenPtr + i) == '\'') {
+              i += 1;
+            }
+          }
+          cur += count + 2;
+        }
+      }
+    } else {
+      int count = 1;
+      ++cur;
+      while (cur < end && *startTokenPtr == *cur) {
+        ++count;
+        ++cur;
+      }
+      switch (*startTokenPtr) {
+        case 'G':
+          builder.appendEra();
+          break;
+        case 'C':
+          builder.appendCenturyOfEra(count);
+          break;
+        case 'Y':
+          builder.appendYearOfEra(count);
+          break;
+        case 'x':
+          builder.appendWeekYear(count);
+          break;
+        case 'w':
+          builder.appendWeekOfWeekYear(count);
+          break;
+        case 'e':
+          builder.appendDayOfWeek1Based(count);
+          break;
+        case 'E':
+          builder.appendDayOfWeekText(count);
+          break;
+        case 'y':
+          builder.appendYear(count);
+          break;
+        case 'D':
+          builder.appendDayOfYear(count);
+          break;
+        case 'M':
+          if (count <= 2) {
+            builder.appendMonthOfYear(count);
+          } else {
+            builder.appendMonthOfYearText(count);
+          }
+          break;
+        case 'd':
+          builder.appendDayOfMonth(count);
+          break;
+        case 'a':
+          builder.appendHalfDayOfDay();
+          break;
+        case 'K':
+          builder.appendHourOfHalfDay(count);
+          break;
+        case 'h':
+          builder.appendClockHourOfHalfDay(count);
+          break;
+        case 'H':
+          builder.appendHourOfDay(count);
+          break;
+        case 'k':
+          builder.appendClockHourOfDay(count);
+          break;
+        case 'm':
+          builder.appendMinuteOfHour(count);
+          break;
+        case 's':
+          builder.appendSecondOfMinute(count);
+          break;
+        case 'S':
+          builder.appendFractionOfSecond(count);
+          break;
+        case 'z':
+          builder.appendTimeZone(count);
+          break;
+        case 'Z':
+          builder.appendTimeZoneOffsetId(count);
+          break;
+        default:
+          if (isalpha(*startTokenPtr)) {
+            VELOX_UNSUPPORTED("Specifier {} is not supported.", *startTokenPtr);
+          } else {
+            builder.appendLiteral(startTokenPtr, cur - startTokenPtr);
+          }
+          break;
+      }
+    }
+  }
+  DateTimeFormatterType type = lenient ? DateTimeFormatterType::LENIENT_SIMPLE
+                                       : DateTimeFormatterType::STRICT_SIMPLE;
```

Comment:
Wondering where the LENIENT_SIMPLE and STRICT_SIMPLE will be used. I only find their definitions in this PR but no usage.

Reply:
It will be used when lenient and non-lenient modes have different code branches. For example, if DateTimeFormatterType is LENIENT_SIMPLE, the helper function "daysSinceEpochFromWeekOfMonthDate" will be called with "lenient=true" otherwise with "lenient=false"